backend/src/main.rs

```diff
@@ -9,6 +9,12 @@ use std::{
 
 const INITIAL_TOKENS: Amount = 1000;
 
+// The maximum number of answers that can be provided per task.
+const MAX_NUMBER_ANSWERS: usize = 10; // TODO: or should we use amount only for token amounts?
```

Comment:
The TODO seems obsolete as you are using `usize`. Fwiw, yeah I think we should `Amount` to represent an amount of tokens, not for other things.

Reply:
yes, resolved this todo in the meantime, sorry